scheduler/src/cook/cors.clj

```diff
@@ -0,0 +1,59 @@
+(ns cook.cors)
+
+(defn preflight?
+  [req]
+  (= :options (:request-method req)))
+
+(defn same-origin?
+  "Returns true if the request is from the same origin as the provided origin header"
+  [{:keys [headers scheme]}]
+  (let [{:strs [host origin]} headers]
+    (when (and host origin scheme)
+      (= origin (str (name scheme) "://" host)))))
+
+(defn request-allowed?
+  "Returns true if the request is either from the same origin or matches a pattern in cors-origins"
```

Comment:
Nitpick: the docstring should mention that it expects `origin` to be non-nil.

Reply:
Done